### PR TITLE
Fixes accumulating ability icons

### LINKED ISSUE #7441: Picking up some items makes you accumulate ability icons
### Environment
- Game Version: UnityStationDevelop-21091918

### Detailed description of bug

image
Items that cause this:
- any bag
- bag of dice
- trash bag

Picking up same item does not duplicate ability. Each item only adds one.

### Steps to reproduce

1. Pick up any listed item
2. Drop it

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [Fix] - Fixed accumulating ability icons (Fixes #7441)

### CODE REVIEW

**Comment by riquelme-A** [comment]
so this means it will still pile up but it is hidden?

**Comment by MaxIsJoe** [comment]
> so this means it will still pile up but it is hidden?

Yeah

**Review by Bod9001 — DISMISSED** [review]
need to make  This server side some point

**Comment by corp-0** [comment]
I'm not sure I'm happy with this approach to fix the bug

**Comment by CdrCross** [comment]
Once we've moved intents off the number keys, I'm eyeing a chance to redesign the ability bar entirely, to act more like it does in an MMO or ARPG, and crucially to eat less space. @corp-0

**Comment by MaxIsJoe** [comment]
> I'm not sure I'm happy with this approach to fix the bug

I'll look for the function that spawns them then and do the check there instead and delete them if they are not available.

Though an Idea that came to my mind would be action usage counters, so imagine an item that has 5 usages and it generates 5 action buttons, we can hide 4 and make one display a counter of how many usages it has and delete the other buttons once they are no longer needed.

**Comment by CdrCross** [comment]
> > I'm not sure I'm happy with this approach to fix the bug
> 
> I'll look for the function that spawns them then and do the check there instead and delete them if they are not available.
> 
> Though an Idea that came to my mind would be action usage counters, so imagine an item that has 5 usages and it generates 5 action buttons, we can hide 4 and make one display a counter of how many usages it has and delete the other buttons once they are no longer needed.

A charge counter is a good idea, but make sure to give it context for the user. We don't need yet another unknowable number floating around. I would recommend either "x uses" or "x/x".

**Comment by MaxIsJoe** [comment]
> I'm not sure I'm happy with this approach to fix the bug

Done, it won't spawn anymore now if there's another button with the same action data.
Review pls

File: UnityProject/Assets/Scripts/UI/Core/Action/UIActionManager.cs [hunk 1]
```diff
@@ -186,6 +186,12 @@ public static void SetCooldown(IActionGUI iActionGUI, float cooldown, GameObject
 
 		public static void Show(IActionGUI iActionGUI)
 		{
+			foreach (var actionButton in Instance.DicIActionGUI)
+			{
+				//If there is a duplicate of this button, don't spawn it!
+				if (actionButton.Value.ActionData != iActionGUI.ActionData) continue;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Core/Action/UIActionManager.cs]: Isn't it simpler to invert this logic so no continue is needed?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Core/Action/UIActionManager.cs]: Done

**Review by corp-0 — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**